mutiny-core/src/nostr/mod.rs

```diff
@@ -148,6 +147,41 @@ impl<S: MutinyStorage> NostrManager<S> {
         Ok(nwc_profile)
     }
 
+    pub fn set_nwc_profile_budget(
+        &self,
+        profile_index: u32,
+        budget_sats: u64,
+        budget_period_secs: u64,
+        single_max_sats: Option<u64>,
+    ) -> Result<NwcProfile, MutinyError> {
```

Comment:
What if we want to remove it from budget to not budget? Can the update method be more generic? 

It would be great if people could set whether or not mutiny+ is auto approval as well and they could convert it to manual or disable at any time.

Reply:
we have `edit_nwc_profile` where we could just overwrite the spending conditions. Not sure how to make it much easier without making a function for every spending condition